src/domain/bitcoin/onchain/tx-filter.ts

```diff
@@ -14,7 +14,10 @@ export const TxFilter = ({
       if (!!confirmationsLessThan && confirmations >= confirmationsLessThan) {
         return false
       }
-      if (!!addresses && !outputAddresses.some((addr) => addresses.includes(addr))) {
+      if (
+        !!addresses &&
+        !outs.some((out) => out.address != null && addresses.includes(out.address))
```

Comment:
Minor nitpick: `!==` instead of `!=` (We should add an ESLint rule for this).

Bigger point: the name `outs` is too generic? I'd have no idea what it means reading this code with no context.

Reply:
Output is the correct name for this data in the context of a bitcoin transaction.

EG:
- [en.bitcoin.it/wiki: output](https://en.bitcoin.it/wiki/Transaction#Output)
- [rust-bitcoin: output](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/blockdata/transaction.rs#L271)
- [bitcoinjs-lib: outs](https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/types/transaction.d.ts#L26)

All bitcoin transactions are made up of inputs and outputs.